include/UnixStamp.hpp

```diff
@@ -23,19 +25,18 @@
 class UnixStamp
 {
 private:
-    time_t unix;
-    int8_t tz;
-    tm time;
+    unixstamp unix;
+    civil_time time;
 
 public:
-    time_t getUnix();
+    unixstamp getUnix();
     int8_t getTz();
-    tm getTime();
+    civil_time getTime();
 
-    UnixStamp(time_t unix);
-    UnixStamp(time_t unix, int8_t tz);
-    UnixStamp(tm time);
-    UnixStamp(tm time, int8_t tz);
+    UnixStamp(unixstamp unix);
+    UnixStamp(unixstamp unix, int8_t tz);
+    UnixStamp(civil_time time);
+    UnixStamp(civil_time time, int8_t tz);
```

Comment:
I've thought about how time (unix and civil) should be kept. From my point of view unix and civil should be kept in GMT+0, so `tz` now is class field.

Reply:
I kinda liked `tz` in `civil_time`, because there is usually no need in serializing the universal (or GMT) time(stamps) into any form of human readable time ("yyyy-MM-dd hh:mm:ss.sss z"), and when the need arises, it for a likely to display the time to a human user, who would appreciate seeing it in the local time-zone.

This is a common approach: the internal representation of (date)time is strictly in UTC or GMT and only serialization is done for an arbitrary time-zone.

My point was that the parameter should probably be removed (also [here](https://github.com/chifir/UnixStamp/pull/2#discussion_r1780619188)).

But, of course, either way works, whatever you prefer.

Reply:
Your words make me hesitate about the chosen method. May be in the next (or next + 1) iteration this approach will replace existing :)